src/main/java/org/kiwiproject/dropwizard/util/health/HttpConnectionsHealthCheck.java

```diff
@@ -0,0 +1,205 @@
+
+    public HttpConnectionsHealthCheck(MetricRegistry metrics) {
+        this(metrics, DEFAULT_WARNING_THRESHOLD);
+    }
+
+    public HttpConnectionsHealthCheck(MetricRegistry metrics, double warningThreshold) {
+        this.metrics = requireNotNull(metrics);
+
+        checkArgument(warningThreshold > 0.0 && warningThreshold < 100.0, IllegalArgumentException.class,
+                "warningThreshold must be more than 0 and less than 100");
+
+        this.warningThreshold = warningThreshold;
+    }
+
+    @Override
+    protected Result check() {
+        var httpMetrics = metrics.getGauges((name, metric) -> name.startsWith(HTTP_CONN_MANAGER_GAUGE_PREFIX));
+        var clientNames = findHttpClientMetricNames(httpMetrics);
+
+        LOG.trace("Client names: {}", clientNames);
+
+        if (clientNames.isEmpty()) {
+            return newHealthyResult("No HTTP clients found with metrics");
+        }
+
+        var clientHealth = getClientHealth(httpMetrics, clientNames);
+        return determineResult(clientHealth, clientNames.size());
+    }
+
+    @SuppressWarnings("rawtypes")
+    private static Set<String> findHttpClientMetricNames(SortedMap<String, Gauge> httpMetrics) {
+        return httpMetrics.keySet().stream()
+                .map(HttpConnectionsHealthCheck::clientNameFrom)
+                .collect(toSet());
+    }
+
+    private static String clientNameFrom(String gaugeName) {
+        return gaugeName.substring(START_INDEX, gaugeName.lastIndexOf('.'));
+    }
+
+    @SuppressWarnings("rawtypes")
+    private Map<ClientConnectionInfo.HealthStatus, List<ClientConnectionInfo>> getClientHealth(
+            SortedMap<String, Gauge> httpMetrics, Set<String> clientNames) {
+
+        return clientNames.stream()
+                .map(clientName -> getClientConnectionInfo(httpMetrics, clientName))
+                .collect(groupingBy(ClientConnectionInfo::getHealthStatus));
+    }
+
+    @SuppressWarnings("rawtypes")
+    private ClientConnectionInfo getClientConnectionInfo(SortedMap<String, Gauge> httpMetrics, String clientName) {
+        var leasedConnectionsGauge = httpMetrics.get(leasedConnectionsGaugeName(clientName));
+        var maxConnectionsGauge = httpMetrics.get(maxConnectionsGaugeName(clientName));
+
+        var leasedConnections = (int) leasedConnectionsGauge.getValue();
+        var maxConnections = (int) maxConnectionsGauge.getValue();
+
+        var connectionInfo = ClientConnectionInfo.builder()
+                .clientName(clientName)
+                .leased(leasedConnections)
+                .max(maxConnections)
+                .warningThreshold(warningThreshold)
+                .build();
+
+        LOG.trace("{}: {} of {} leased ({}%)", clientName, leasedConnections, maxConnections,
+                connectionInfo.percentLeased);
+
+        return connectionInfo;
+    }
+
+    private Result determineResult(Map<ClientConnectionInfo.HealthStatus, List<ClientConnectionInfo>> clientHealth,
```

Comment:
nit: move ` List<ClientConnectionInfo>> clientHealth` to separate line.

Reply:
Even though it is part of the generic Map definition?

Reply:
Oops my bad. No.